rustls/src/client/common.rs

```diff
@@ -58,15 +57,13 @@ impl ServerKXDetails {
 pub struct HandshakeDetails {
     pub resuming_session: Option<persist::ClientSessionValue>,
     pub session_id: SessionID,
-    pub dns_name: webpki::DNSName,
 }
 
 impl HandshakeDetails {
-    pub fn new(host_name: webpki::DNSName) -> HandshakeDetails {
+    pub fn new() -> HandshakeDetails {
```

Comment:
I think I'd prefer having a separate commit (or PR) for threading the `dns_name` through the handshake states (recommended process: manually edit the diff to ditch the `HelloData`-related chunks).

Reply:
I'm not attached to the approach here, but do you want `dns_name` passed a separate parameter alongside `extra_exts` in the early parts of the handshake?

Reply:
Sorry -- no. Just, in terms of reviewing I'd prefer to separate the commit threading the `dns_name` all the way through the handshake states from the commit that introduces `HelloData` (including packaging up the `dns_name` in it for some of the handshake states).

Reply:
OK, but if I remove `dns_name` from `HandshakeDetails`, it has to go somewhere in the early parts of the handshake. Sorry if I'm being dense here. I see two options that might be easier to review than what's here: 

1) thread it through as a separate parameter alongside `extra_exts` early on, the later parts of the handshake would stay the same.

2) Cut this patch down a bit to use a private Host struct instead of a trait, leaving the public API the same, but bundling `extra_exts` and `dns_name` together.

Reply:
Yeah, I meant the first option. Maybe separate commits to thread dns_name and extra_exts?

Reply:
OK, done.